conductor/blueprints/search/models.py

```diff
@@ -0,0 +1,57 @@
+import os
+import json
+
+from elasticsearch import Elasticsearch
+from elasticsearch.exceptions import NotFoundError
+
+_engine = None
+
+ENABLED_SEARCHES = {
+    'user': {
+        'index': 'users',
+        'doc_type': 'user_profile',
+        '_source': ['idhash', 'name', 'avatar_url', 'datasets']
+    },
+    'package': {
+        'index': 'packages',
+        'doc_type': 'package',
+        '_source': ['id', 'model', 'package', 'origin_url']
+    }
+}
+
+def _get_engine():
+    global _engine
+    if _engine is None:
+        es_host = os.environ['OS_ELASTICSEARCH_ADDRESS']
+        _engine = Elasticsearch(hosts=[es_host], use_ssl='https' in es_host)
+    return _engine
+
+
+def query(kind, size=100, **kw):
+    kind_params = ENABLED_SEARCHES.get(kind)
+    if kind_params is None:
+        return None
+    try:
+        if type(size) is list:
+            size = size[0]
+        api_params = {'size': int(size)}
+        api_params.update(kind_params)
+        q_str = ") AND (".join([" OR ".join(["{0}:{1}".format(k, json.dumps(json.loads(v)))
+                                             for v in v_arr])
+                                for k, v_arr in kw.items()
+                                if k.split('.')[0] in kind_params['_source']])
+        if len(q_str) == 0:
+            q_str = "{0}:*".format(kind_params['_source'][0])
+        else:
+            q_str = "(" + q_str + ")"
+        api_params['q'] = q_str
+        ret = _get_engine().search(**api_params)
+        if ret.get('hits') is not None:
+            return [ hit['_source'] for hit in ret['hits']['hits'] ]
+        return None
+    except NotFoundError:
+        return None
+    except json.JSONDecodeError:
+        return None
+    except ValueError:
+        return None
```

Comment:
This is more about programming styles than anything else, so feel free to ignore.

Given that you're simply ignoring all these exceptions, I would merge them all in a single `except` clause. I would also remove the `return None` on line 51. All in all, apart from validating that the received params are good (as you do on lines 32~33), I prefer to, when possible, keep a single return point in every function. I feel like it's easier to read that way, but again, it's a matter of personal preferences.